### PR TITLE
User Story 2 - Add contact info

### PR DESCRIPTION
**Pull request by YuriPadilla**

### CODE REVIEW

File: components/ContactInfo/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          <article>Mobile: (+53) 5 391 2608</article>
+          <article>Land line: (+53) 7 867 6782</article>
+          <p>email: yuripadilla017@gmail.com</p>
+        </div>
+        <StyledButton
+          type="button"
+          onClick={() => router.push("/ContactUsPage")}
+        >
+          Contact us
+        </StyledButton>
```
Comment (F-Kirchhoff) [inline components/ContactInfo/index.js:24]: please use the next Link component for this usecase instead. This is better for folks who use screenreaders. You don't need the router hook then.

File: components/ContactInfo/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  return (
+    <>
+      <h2>Contact information</h2>
+      <section>
+        <div>
+          <article>
+            61 Consulado St. between Genios and Refugio St. Centro Habana, Cuba.
+          </article>
+          <article>Mobile: (+53) 5 391 2608</article>
+          <article>Land line: (+53) 7 867 6782</article>
```
Comment (F-Kirchhoff) [inline components/ContactInfo/index.js:16]: These are not articles, but either spans or p tags.

File: components/ContactUsForm/index.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            rows="8"
+            cols="40"
+            placeholder="Write your message here..."
+          />
+        </label>
+        <div>
+          <StyledButton type="submit">Send</StyledButton>
+          <StyledButton type="button" onClick={() => router.push("/")}>
+            ← Home
+          </StyledButton>
```
Comment (F-Kirchhoff) [inline components/ContactUsForm/index.js:37]: Same here.

File: components/ContactUsForm/index.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        <label htmlFor="name">
+          *Name:
+          <br />
+          <input type="text" id="name" name="name" required />
+        </label>
+        <label htmlFor="email">
+          *Email:
+          <br />
+          <input type="email" id="email" name="email" required />
+        </label>
```
Comment (F-Kirchhoff) [inline components/ContactUsForm/index.js:21]: Please don't nest the input in the label. Screen readers read all the content of the Label tag, also html. That is pretty annoying if the input tag is read out to you every time you use the form. You already have a htmlFor property and an id, so you can simply move the input next to its label.

File: components/ContactUsForm/index.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        <label htmlFor="message">
+          *Message:
+          <br />
```
Comment (F-Kirchhoff) [inline components/ContactUsForm/index.js:24]: please remove these br tags and use CSS to create a gap between the elements.

File: components/ToastNotification/index.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export default function ToastNotification({ toastAction }) {
+  return (
+    <button className={`toast ${toastAction}`}>Your message was sent</button>
```
Comment (F-Kirchhoff) [inline components/ToastNotification/index.js:3]: Please don' use class names alongside with styled Components. We write the CSS purely in styled components.

Comment (F-Kirchhoff) [reply components/ToastNotification/index.js:3]: you can pass props to your styled component to achieve the same as you do with the toastAction

File: pages/ContactUsPage/index.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  function exitToast() {
+    setToastAction("exit");
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    console.log(data);
```
Comment (F-Kirchhoff) [inline pages/ContactUsPage/index.js:18]: remove all console.logs before pushing 🕵️

File: styles/styles.css [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+.toast.enter {
+  transform: translateX(0);
+  transition: transform 1s cubic-bezier(0, 0.79, 1, 1.02);
+}
+
+.toast.exit {
+  transform: translateX(150%);
+  transition: transform 1s cubic-bezier(0, -0.03, 1, 0.41);
+}
```
Comment (F-Kirchhoff) [inline styles/styles.css:24]: you need to solve this with styled components. You can use toastAction as a prop inside your ToastNotification styled Component to add different stylings.

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Yuri, 
besides the CSS issues and the Links disguised as buttons you are ready to go. Good work!

File: pages/index.js [hunk 9]
```diff
@@ -10,6 +11,7 @@ export default function HomePage() {
       <p>Home</p>
```
Comment (Long-HaHoang) [inline pages/index.js:11]: What's the point of this paragraph?
Do you really need to tell the user, that this is the home page?
It's already your landing page.
Besides if you want to keep it I suggest you to use a `<h2>` here.

File: pages/ContactUsPage/index.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    setToastAction("enter");
+    setTimeout(exitToast, 3000);
+  }
+
+  return (
+    <>
+      <Header />
+      <p>
+        <Link href="/">Home</Link>→Contact us
+      </p>
```
Comment (Long-HaHoang) [inline pages/ContactUsPage/index.js:30]: Now you have two ways on same page to move back to the home page.
Besides that, you already have a heading for "Contact us" in your form.

**Review by Long-HaHoang — CHANGES_REQUESTED** [review]
Hi Yuri,
Your links are still inside of buttons.
And your input fields need some validations.

File: components/ContactUsForm/index.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    <StyledForm onSubmit={handleSubmit}>
+      <h2>Contact us</h2>
+
+      <label htmlFor="name">*Name:</label>
+      <input type="text" id="name" name="name" required />
+
+      <label htmlFor="email">*Email:</label>
+      <input type="email" id="email" name="email" required />
+
+      <label htmlFor="message">*Message:</label>
```
[the remark this answers is not in the document]
Comment (YuriPadilla) [reply components/ContactUsForm/index.js:17]: Hi Long!
Well... yes, I just wanted to indicate the user that this input is required.

File: components/ContactUsForm/index.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+//import { useRouter } from "next/router";
+import Link from "next/link";
+
+export default function ContactUsForm({ handleSubmit }) {
+  return (
+    <StyledForm onSubmit={handleSubmit}>
+      <h2>Contact us</h2>
+
+      <label htmlFor="name">*Name:</label>
+      <input type="text" id="name" name="name" required />
```
[the remark this answers is not in the document]
Comment (YuriPadilla) [reply components/ContactUsForm/index.js:12]: A pattern to the name... maybe later...

File: pages/index.js (hunk 9 above)
Comment (YuriPadilla) [reply pages/index.js:11]: The point is to fallow the same pattern as in other pages... and I use it to show the history of the links. A `<h2>` is not properly here.

Comment (YuriPadilla) [reply pages/index.js:11]: I think it is properly to keep it.

File: pages/ContactUsPage/index.js (hunk 10 above)
Comment (YuriPadilla) [reply pages/ContactUsPage/index.js:30]: I think it's OK so...

File: components/ContactInfo/index.js [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+            Mobile: (+53) 5 391 2608
+            <br />
+            Land line: (+53) 7 867 6782
+            <br />
+            email: yuripadilla017@gmail.com
+          </p>
+        </article>
+        <StyledButton type="button">
+          <Link href="/ContactUsPage">Contact us</Link>
+        </StyledButton>
```
[the remark this answers is not in the document]
Comment (YuriPadilla) [reply components/ContactInfo/index.js:23]: Good Idea, Thanks!!!  :)

File: components/ContactUsForm/index.js [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+      <input type="email" id="email" name="email" required />
+
+      <label htmlFor="message">*Message:</label>
+      <textarea
+        id="message"
+        name="message"
+        rows="8"
+        cols="40"
+        placeholder="Write your message here..."
+      />
```
[the remark this answers is not in the document]
Comment (YuriPadilla) [reply components/ContactUsForm/index.js:24]: You are right! ...I thought I had it required.

**Comment by YuriPadilla** [comment]
Thank you Long-HaHoang and F-Kirchhoff for reviewing my code!!!!! it was really helpful. Hope to get more reviews from you all...!!!

**Pull request merged**